doc/how-to/app/app_roles.rst

```diff
@@ -0,0 +1,476 @@
+.. _application_roles:
+
+Application roles
+=================
+
+An application role is a Lua module that implements specific functions or logic.
+You can enable or turn off a particular role for certain instances in a :ref:`configuration <configuration>` without restarting these instances.
+A role is run when a configuration is loaded or reloaded.
+
+Roles can be divided into the following groups:
+
+-   Roles provided by Tarantool.
+    For example, the ``config.storage`` role can be used to make a Tarantool replica set act as a :ref:`configuration storage <centralized_configuration_storage_set_up_tarantool>`.
+-   Roles provided by third-party Lua modules.
+    For example, the `CRUD <https://github.com/tarantool/crud>`__ module provides the ``roles.crud-storage`` and ``roles.crud-router`` roles that enable CRUD operations in a sharded cluster.
+-   Custom roles that are developed as a part of a cluster application.
+    For example, you can create a custom role to define a stored procedure or implement a supplementary service, such as an email notifier or a replicator.
+
+This section describes how to develop custom roles.
+To learn how to enable and configure roles, see :ref:`configuration_application_roles`.
+
+..  NOTE::
+
+    Don't confuse application roles with other role types:
+
+    -   A role is a container for privileges that can be granted to users. Learn more in :ref:`access_control_concepts_roles`.
+    -   A role of a replica set in regard to sharding. Learn more in :ref:`vshard_config_sharding_roles`.
+
+
+
+
+.. _roles_create_custom_role_config:
+
+Providing a role configuration
+------------------------------
+
+A custom role can be configured in the same way as roles provided by Tarantool or third-party Lua modules.
+You can learn more from :ref:`configuration_application_roles`.
+
+This example shows how to enable and configure the ``greeter`` role, which is implemented in the next section:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/application_role_cfg/config.yaml
+    :language: yaml
+    :start-at: instance001
+    :dedent:
+
+The role's configuration provided in ``roles_cfg`` can be accessed when :ref:`validating <roles_create_custom_role_validate>` and :ref:`applying <roles_create_custom_role_apply>` this configuration.
+
+Given that a role is a :ref:`Lua module <app_server-modules>`, a role's name is passed to ``require()`` to obtain the module.
+When :ref:`developing an application <admin-instance_config-develop-app>`, you can place a file with a role's code next to a cluster's configuration file.
+
+
+
+.. _roles_create_custom_role:
+
+Creating a custom role
+----------------------
+
+.. _roles_create_custom_role_overview:
+
+Overview
+~~~~~~~~
+
+Creating a custom role includes the following steps:
+
+1.  Define a function that validates a role's configuration.
```

Comment:
I'd provide the function names explicitly in this list.

Reply:
This list give a general idea about role's functions. The code snippet after this lists the exact names.